libs/metagen/src/client_rs/static/client.rs

```diff
@@ -0,0 +1,1935 @@
+        let mut data = data.into_iter();
+        serde_json::from_value(data.next().unwrap())
+    }
+}
+impl<Out> ToQueryDoc for QueryNode<Out> {}
+impl<Out> ToSelectDoc for MutationNode<Out>
+where
+    Out: serde::de::DeserializeOwned,
+{
+    type Out = Out;
+
+    fn to_select_doc(self) -> Vec<SelectNodeErased> {
+        vec![self.0]
+    }
+
+    fn parse_response(data: Vec<serde_json::Value>) -> Result<Self::Out, serde_json::Error> {
+        let mut data = data.into_iter();
+        serde_json::from_value(data.next().unwrap())
+    }
+}
+impl<Out> ToMutationDoc for MutationNode<Out> {}
+
+#[macro_export]
+macro_rules! impl_for_tuple {
+    ($($idx:tt $ty:tt),+) => {
+        impl<$($ty,)+> ToSelectDoc for ($(QueryNode<$ty>,)+)
+            where $($ty: serde::de::DeserializeOwned,)+
+        {
+            type Out = ($($ty,)+);
+
+            fn to_select_doc(self) -> Vec<SelectNodeErased> {
+                vec![
+                    $(self.$idx.0,)+
+                ]
+            }
+            fn parse_response(data: Vec<serde_json::Value>) -> Result<Self::Out, serde_json::Error> {
+                let mut data = data.into_iter();
+                let mut next = move |_idx| data.next().unwrap();
+                Ok((
+
+                    $(serde_json::from_value(next($idx))?,)+
+                ))
+            }
+        }
+        impl<$($ty,)+> ToSelectDoc for ($(MutationNode<$ty>,)+)
+            where $($ty: serde::de::DeserializeOwned,)+
+        {
+            type Out = ($($ty,)+);
+
+            fn to_select_doc(self) -> Vec<SelectNodeErased> {
+                vec![
+                    $(self.$idx.0,)+
+                ]
+            }
+            fn parse_response(data: Vec<serde_json::Value>) -> Result<Self::Out, serde_json::Error> {
+                let mut data = data.into_iter();
+                let mut next = move |_idx| data.next().unwrap();
+                Ok((
+
+                    $(serde_json::from_value(next($idx))?,)+
+                ))
+            }
+        }
+
+        impl<$($ty,)+> ToQueryDoc for ($($ty,)+)
+        where
+            $($ty: ToQueryDoc,)+
+        {}
+
+        impl<$($ty,)+> ToMutationDoc for ($($ty,)+)
+        where
+            $($ty: ToMutationDoc,)+
+        {}
+    };
+}
+
+impl_for_tuple!(0 N0);
+impl_for_tuple!(0 N0, 1 N1);
+impl_for_tuple!(0 N0, 1 N1, 2 N2);
+impl_for_tuple!(0 N0, 1 N1, 2 N2, 3 N3);
+impl_for_tuple!(0 N0, 1 N1, 2 N2, 3 N3, 4 N4);
+impl_for_tuple!(0 N0, 1 N1, 2 N2, 3 N3, 4 N4, 5 N5);
+impl_for_tuple!(0 N0, 1 N1, 2 N2, 3 N3, 4 N4, 5 N5, 6 N6);
+impl_for_tuple!(0 N0, 1 N1, 2 N2, 3 N3, 4 N4, 5 N5, 6 N6, 7 N7);
+impl_for_tuple!(0 N0, 1 N1, 2 N2, 3 N3, 4 N4, 5 N5, 6 N6, 7 N7, 8 N8);
+impl_for_tuple!(0 N0, 1 N1, 2 N2, 3 N3, 4 N4, 5 N5, 6 N6, 7 N7, 8 N8, 9 N9);
+impl_for_tuple!(0 N0, 1 N1, 2 N2, 3 N3, 4 N4, 5 N5, 6 N6, 7 N7, 8 N8, 9 N9, 10 N10);
+impl_for_tuple!(0 N0, 1 N1, 2 N2, 3 N3, 4 N4, 5 N5, 6 N6, 7 N7, 8 N8, 9 N9, 10 N10, 11 N11);
+
```

Comment:
Why 11? Seems arbitrary..

Reply:
No good reason. Anecdotal but 12 tuples is what I've usually observed in tupl impls in crate sources I've read in the past.